src/Compilers/CSharp/Portable/Binder/DecisionDagBuilder.cs

```diff
@@ -316,24 +318,29 @@ Tests scanAndSimplify(Tests tests)
             RoslynDebug.Assert(iTupleType.Name == "ITuple");
             var tests = ArrayBuilder<Tests>.GetInstance(4 + patternLength * 2);
 
+            output = input = MakeConvertToType(input, syntax, iTupleType, false, tests);
```

Comment:
❓ There's another subtlety with the unification: currently if we're matching an ITuple directly, we still emit a type check. To make this consistent across all orderings we may want to stop doing that which could affect subsumption results in some cases.